From 9aa94138d2c2bb1c0cd456b24e7e7efebacf21e3 Mon Sep 17 00:00:00 2001
From: Tien Le <tien.le.xw@renesas.com>
Date: Wed, 12 May 2021 15:19:00 +0700
Subject: [PATCH 170/180] driver: thermal: rzg2l_thermal: Modify
 rzg2l_thermal_round() function

This pacth modify rzg2l_thermal_round() function. For example:
We have mCelsius = 35231 (35.231 Celsius) , after apply formula.
We have result is mCelsius = 35500 (35.5 Celsius)

Signed-off-by: Tien Le <tien.le.xw@renesas.com>
---
 drivers/thermal/rzg2l_thermal.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/thermal/rzg2l_thermal.c b/drivers/thermal/rzg2l_thermal.c
index c23ae8c..6bae87b 100644
--- a/drivers/thermal/rzg2l_thermal.c
+++ b/drivers/thermal/rzg2l_thermal.c
@@ -45,6 +45,7 @@
 #define INT(x)	((x) * 1000000)
 #define MCELSIUS(temp)	((temp) * 1000)	/* mili Celsius */
 #define CAP_TIMES	8	/* Capture  times */
+#define RZG2L_THERMAL_GRAN	500 /* mili Celsius */
 
 typedef enum
 {
@@ -80,13 +81,15 @@ static inline void rzg2l_thermal_write(struct rzg2l_thermal_tsc *tsc,
 	iowrite32(data, tsc->base + reg);
 }
 
-static int rzg2l_thermal_round(int temp)
+static int rzg2l_thermal_round(int mCelsius)
 {
-		int result;
+	int result, round_offs;
 
-		result = temp / 10;
+	round_offs = mCelsius >= 0 ? RZG2L_THERMAL_GRAN / 2 :
+		-RZG2L_THERMAL_GRAN / 2;
+	result = (mCelsius + round_offs) / RZG2L_THERMAL_GRAN;
 
-		return result + TEMP_25;
+	return result * RZG2L_THERMAL_GRAN;
 }
 
 static int rzg2l_thermal_get_temp(void *devdata, int *temp)
-- 
2.7.4

